cpu/cpu.go

```diff
@@ -232,29 +224,33 @@ func (p *Plugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.Metri
 
 // GetConfigPolicy returns config policy
 // It returns error in case retrieval was not successful
-func (p *Plugin) GetConfigPolicy() (*cpolicy.ConfigPolicy, error) {
-	cp := cpolicy.New()
-	rule, _ := cpolicy.NewStringRule("proc_path", false, "/proc")
-	node := cpolicy.NewPolicyNode()
-	node.Add(rule)
-	cp.Add([]string{vendor, fs, pluginName}, node)
-	return cp, nil
+func (p *Plugin) GetConfigPolicy() (plugin.ConfigPolicy, error) {
+	policy := plugin.NewConfigPolicy()
+	policy.AddNewStringRule([]string{vendor, fs, Name}, "proc_path", false, plugin.SetDefaultString("/proc"))
+
+	return *policy, nil
 }
 
-//Meta returns meta data for plugin
-func Meta() *plugin.PluginMeta {
-	return plugin.NewPluginMeta(
-		pluginName,
-		version,
-		pluginType,
-		[]string{},
-		[]string{plugin.SnapGOBContentType},
-		plugin.ConcurrencyCount(1))
+// New creates instance of interface info plugin
+func New() *Plugin {
+	host, err := os.Hostname()
+	if err != nil {
+		host = "localhost"
+	}
+	p := &Plugin{
+		host:      host,
+		proc_path: cpuInfo,
+	}
+	return p
 }
 
-func (p *Plugin) init(cfg map[string]ctypes.ConfigValue) error {
-	if procPath, ok := cfg["proc_path"]; ok {
-		p.proc_path = procPath.(ctypes.ConfigValueStr).Value + "/stat"
+func (p *Plugin) init(cfg plugin.Config) error {
+	if _, ok := cfg["proc_path"]; ok {
```

Comment:
Without this check, I was noticing quite some assertion failures, so I added it back.